src/main/java/com/selfxdsd/todocli/TodoParser.java

```diff
@@ -77,13 +73,16 @@ public TodoParser() {
      * @param path Path to the file being parsed.
      * @return List of found TODOs.
      * @throws IOException If something goes wrong.
+     * @checkstyle ExecutableStatementCount (50 lines)
      */
     public List<Todo> parse(final String path) throws IOException {
         List<String> lines = Files.readAllLines(Paths.get(path));
         List<Todo> todos = new ArrayList<>();
+        final AtomicInteger currentIndex = new AtomicInteger();
```

Comment:
@amihaiemil I've used a here an atomic int to be able to pass currentIndex to `parseMultiLineTodo` as a mutable ref. Before `currentIndex` was a class member and thus not thread safe.